src/internal/events/events.go

```diff
@@ -165,8 +169,15 @@ func storageID(s storage.Storage) string {
 	return id
 }
 
-func repoHash(s storage.Storage) string {
-	return md5HashOf(storageID(s))
+// NewRepoID generates a new unique repository id hash.
+// Repo ids should only be generated once per repository,
+// and must be stored after that.
+func NewRepoID(s storage.Storage) string {
+	return repoHash(s, uuid.NewString())
+}
+
+func repoHash(s storage.Storage, uid string) string {
```

Comment:
this should take a storageID as a parameter instead of asking storage to fetch it